application/src/test/java/org/thingsboard/server/system/BaseHttpDeviceApiTest.java

```diff
@@ -74,6 +79,54 @@ public void testGetAttributes() throws Exception {
         doGetAsync("/api/v1/" + deviceCredentials.getCredentialsId() + "/attributes?clientKeys=keyA,keyB,keyC").andExpect(status().isOk());
     }
 
+    @Test
+    public void testReplyToCommandWithLargeResponse() throws Exception {
+        String errorResponse = doPost("/api/v1/" + deviceCredentials.getCredentialsId() + "/rpc/5",
+                JacksonUtil.toString(createRpcResponsePayload(10001)),
+                String.class,
+                status().is4xxClientError());
+        assertThat(errorResponse).contains("Payload size exceeds the configured maximum");
+
+        doPost("/api/v1/" + deviceCredentials.getCredentialsId() + "/rpc/5",
+                JacksonUtil.toString(createRpcResponsePayload(10000)),
+                String.class,
+                status().isOk());
+    }
+
+    @Test
+    public void testPostRpcRequestWithLargeResponse() throws Exception {
+        String errorResponse = doPost("/api/v1/" + deviceCredentials.getCredentialsId() + "/rpc",
+                JacksonUtil.toString(createRpcRequestPayload(10001)),
+                String.class,
+                status().is4xxClientError());
+        assertThat(errorResponse).contains("Payload size exceeds the configured maximum");
+
+        doPost("/api/v1/" + deviceCredentials.getCredentialsId() + "/rpc",
+                JacksonUtil.toString(createRpcRequestPayload(10000)),
+                String.class,
+                status().isOk());
+    }
+
+    private ObjectNode createRpcResponsePayload(int size) {
+        ObjectNode rpcResponseBody = JacksonUtil.newObjectNode();
+        char[] chars = new char[size-13];
+        Arrays.fill(chars, 'a');
+        rpcResponseBody.put("result", new String(chars));
+        return rpcResponseBody;
+    }
+
+    private ObjectNode createRpcRequestPayload(int size) {
+        ObjectNode rpcResponseBody = JacksonUtil.newObjectNode();
+        char[] chars = new char[size-38];
+        Arrays.fill(chars, 'a');
+        rpcResponseBody.put("method", "get");
+        ObjectNode params = JacksonUtil.newObjectNode();
+        params.put("value", new String(chars));
+        rpcResponseBody.set("params", params);
```

Comment:
kind of overengineered.
Lets compose some simple string that are easy to read and understand like
```
{"method":"get","params":{"value":"aaaaaaaaaaaaaa"}}
```
the only "aaaaaaaa" can be generated. other just a static string . much clear as for me